pkg/services/pluginsintegration/clientmiddleware/caching_middleware.go

```diff
@@ -77,7 +80,10 @@ func (m *CachingMiddleware) QueryData(ctx context.Context, req *backend.QueryDat
 
 	// Update the query cache with the result for this metrics request
 	if err == nil && cr.UpdateCacheFn != nil {
-		cr.UpdateCacheFn(ctx, resp)
+		// Check if it is a still running async query, if it is don't cache it
+		if shouldCacheQuery(resp) {
```

Comment:
Just occurred to me, I think we want to check the feature toggle state here, right?
```
if !features.IsEnabled(featuremgmt.FlagAwsAsyncQueryCaching) {
  cr.UpdateCacheFn(ctx, resp)
} else if shouldCacheQuery(resp) {
  cr.UpdateCacheFn(ctx, resp)
}
```

Reply:
done